memif-attach/main.go

```diff
@@ -53,7 +68,7 @@ func createMemifSocket(ctx context.Context, conn api.Connection) (socketID uint3
 	MemifSocketFilenameAddDel := memif.MemifSocketFilenameAddDel{
 		IsAdd:          true,
 		SocketID:       1,
-		SocketFilename: "FirstSocketFile",
+		SocketFilename: "/var/run/vpp/memif.sock",
```

Comment:
Try a different filename (still fully qualified, but different)... I suspect that vpp is already creating this as 'socket file 0' and so its not letting you create another one.

Reply:
But should I be able to see the sock file in the path?

Reply:
If I look into /var/run/vpp/, I should see the sock file, right?

Reply:
`FATA[0000] ERROR: connect to VPP master failed: dial unixpacket /var/run/vpp/memif1.sock: connect: no such file or directory`
When trying a different sock name, I still get `no such file` error.

Reply:
@glassxer1 I think vpp doesn't listen on the socket until someone creates a memif interface for that socketid.  Your best bet really is to just try a different name.

Please also note you can `docker exec` into a shell in your docker container and run vppctl commands to see what's going on with the running vpp: https://docs.fd.io/vpp/21.06/d2/d62/clicmd_src_plugins_memif.html

Reply:
Running `show memif` in `vppctl` gave me the following logs:
`vpp# show memif
sockets
  id  listener    filename
  0   no          /run/vpp/memif.sock
  1   no          /var/run/vpp/memif1.sock

interface memif0/0
  socket-id 0 id 0 mode ip
  flags slave zero-copy
  listener-fd 0 conn-fd 0
  num-s2m-rings 0 num-m2s-rings 0 buffer-size 0 num-regions 0
`
So it seems like a memif socket has been created successfully, but net.Dial can not locate the socket file? 
Any other info I can gather by running vppctl?

Reply:
Can you see the file with ls ?

Reply:
Also, are you using 'unixpacket' ad the network type?

Reply:
I can not see the file under `/var/run/vpp/memif1.sock` nor `/run/vpp/memif.sock`. Besides, `unixpacket` is used `socketAddr := "/var/run/vpp/memif.sock"
	conn_client, err := net.Dial("unixpacket", socketAddr)`

Reply:
Try adding logging for your memifCreate so that you capture:
1. The sockID you are passing to vpp
2. The interface swIndex returned.
3. In addition to `show memif` capture `show int`

Reply:
If you get a `show memif` output that has the correct socketID and you still aren't seeing the socketfile, try using

[SwInterfaceSetFlags](https://github.com/edwarnicke/govpp/blob/main/binapi/interface/interface_rpc.ba.go#L34)

To set [Flags to AdminUp](https://github.com/edwarnicke/govpp/blob/97cb6fe890e9ae48c9ff4bf37ed1a0d1e9503bbd/binapi/interface_types/interface_types.ba.go#L31)